src/careamics/lightning/callbacks/prediction_writer_callback.py

```diff
@@ -0,0 +1,242 @@
+"""Module containing PredictionWriterCallback class."""
+
+from pathlib import Path
+from typing import Any, Optional, Sequence, Union
+
+from pytorch_lightning import LightningModule, Trainer
+from pytorch_lightning.callbacks import BasePredictionWriter
+
+from careamics.config.support import SupportedData
+from careamics.file_io import WriteFunc, get_write_func
+from careamics.utils import get_logger
+
+logger = get_logger(__name__)
+
+
+class PredictionWriterCallback(BasePredictionWriter):
+    """
+    A PyTorch Lightning callback to save predictions.
+
+    Parameters
+    ----------
+    save_type : SupportedData or str, default="tiff"
+        The data type to save as, includes custom.
+    save_func : WriteFunc, optional
+        If a known `save_type` is selected this argument is ignored. For a custom
+        `save_type` a function to save the data must be passed. See notes below.
+    save_extension : str, optional
+        If a known `save_type` is selected this argument is ignored. For a custom
+        `save_type` an extension to save the data with must be passed.
+    save_func_kwargs : dict of {str: any}, optional
+        Additional keyword arguments to be passed to the save function.
+    dirpath : pathlib.Path or str, default="predictions"
+        Directory to save outputs to. If `dirpath is not absolute it is assumed to
+        be relative to current working directory. Nested directories will not be
+        automatically created.
+
+    Attributes
+    ----------
+    save_predictions : bool
+        A toggle to optionally switch off prediction saving.
+    save_type : SupportedData
+        The data type of the saved data.
+    save_func : WriteFunc
+        The function for saving data.
+    save_extension: str
+        The extension that will be added to the save paths.
+    save_func_kwargs : dict of {str, any}
+        Additional keyword arguments that will be passed to the save function.
+    dirpath : pathlib.Path
+        The path to the directory where prediction outputs will be saved.
+    """
+
+    def __init__(
+        self,
+        save_type: Union[SupportedData, str] = "tiff",
```

Comment:
Should we use `Literal["tiff", "custom"]` instead of `str`?

Then the SupportedData is not exactly equivalent, so we could also just put `Liberal["array", "tiff", "custom]`.

Reply:
Yes I think maybe for all the user facing functions use `Literal`, have the value converted to `SupportedData` early on, and all internal functions expect `SupportedData` type.

Reply:
That would make a lot of sense, yes!